tcmur_cmd_handler.c

```diff
@@ -144,6 +144,124 @@ static int write_work_fn(struct tcmu_device *dev,
 				block_size * tcmu_get_lba(cmd->cdb));
 }
 
+static void handle_write_same_16_null_cbk(struct tcmu_device *dev,
+					  struct tcmulib_cmd *cmd,
+					  int ret)
+{
+}
+
+static int write_same_16_work_fn(struct tcmu_device *dev,
+				 struct tcmulib_cmd *cmd)
+{
+	struct tcmur_handler *rhandler = tcmu_get_runner_handler(dev);
+	uint32_t block_size = tcmu_get_dev_block_size(dev);
+	uint32_t lba_cnt = tcmu_get_xfer_length(cmd->cdb);
+	uint64_t start_lba = tcmu_get_lba(cmd->cdb);
+	uint64_t end_lba;
+	uint32_t max_lbas, copy_lbas;
+	uint32_t max_sectors;
+	int ret;
+
+	end_lba = start_lba + lba_cnt;
+
+	max_sectors = tcmu_get_attribute(dev, "hw_max_sectors");
```

Comment:
Don't worry about replying late. It is no problem. I am most guilty of slow replies :)

Agree on the bdev info.

I was more asking about what Andy intended vs how you were using it. If I understand your second comment, we agree that hw_max_sectors == max_xfer_length and it should be in block_sized units and we do not have to convert between 512 bytes.

I think the 512b conversion we did in the vpd code was a carry over where in many places in the kernel sectors means 512b units and we have to swap back and forth. The user will not know this, so when they are specifying hw_max_sectors in their target config files, they are just going to specify number of block_sized sectors.

Here is a patchset that fixes that:
https://github.com/open-iscsi/tcmu-runner/pull/147